src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -518,15 +524,56 @@ private void createDir(String dir) {
   private Partitioner newPartitioner(HdfsSinkConnectorConfig config)
       throws ClassNotFoundException, IllegalAccessException, InstantiationException {
 
-    @SuppressWarnings("unchecked")
-    Class<? extends Partitioner> partitionerClass =
-        (Class<? extends Partitioner>) config.getClass(PartitionerConfig.PARTITIONER_CLASS_CONFIG);
+    Partitioner partitioner;
+    try {
+      @SuppressWarnings("unchecked")
+      Class<? extends Partitioner> partitionerClass =
+          (Class<? extends Partitioner>)
+              config.getClass(PartitionerConfig.PARTITIONER_CLASS_CONFIG);
+      partitioner = partitionerClass.newInstance();
+    } catch (ClassCastException e) {
+      @SuppressWarnings("unchecked")
+      Class<? extends io.confluent.connect.storage.partitioner.Partitioner<FieldSchema>>
+          partitionerClass =
+          (Class<? extends io.confluent.connect.storage.partitioner.Partitioner<FieldSchema>>)
+              config.getClass(PartitionerConfig.PARTITIONER_CLASS_CONFIG);
+      partitioner = new PartitionerWrapper(partitionerClass.newInstance());
+    }
 
-    Partitioner partitioner = partitionerClass.newInstance();
     partitioner.configure(new HashMap<>(config.plainValues()));
     return partitioner;
   }
 
+  public static class PartitionerWrapper implements Partitioner {
+    public io.confluent.connect.storage.partitioner.Partitioner<FieldSchema>  partitioner;
```

Comment:
Sorry, one more nit. Should this be `final`? It wasn't really necessary when it was private, but now it seems like it might be necessary.

Reply:
Indeed, makes sense.